avahi fails to build on FreeBSD when `-lintl` isn't explicticly passed via `LDFLAGS`

It was spotted by the CI discussed in https://github.com/avahi/avahi/issues/724

https://github.com/evverx/avahi/actions/runs/18868708024/job/53841696327?pr=3
```sh
/bin/sh ../libtool  --tag=CC   --mode=link clang -I.. '-DDEBUG_TRAP=__asm__("int $3")' -I/usr/local/include -g -O0 -fstack-protector -std=c99 -Wall -W -Wextra -pedantic -pipe -Wformat -Wold-style-definition -Wdeclaration-after-statement -Wfloat-equal -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -fdiagnostics-show-option -Wno-cast-qual -fno-strict-aliasing   -version-info 8:0:1 -L/usr/local/lib -o libavahi-core.la -rpath /usr/local/lib libavahi_core_la-timeeventq.lo libavahi_core_la-iface.lo libavahi_core_la-server.lo libavahi_core_la-entry.lo libavahi_core_la-prioq.lo libavahi_core_la-cache.lo libavahi_core_la-socket.lo libavahi_core_la-response-sched.lo libavahi_core_la-query-sched.lo libavahi_core_la-probe-sched.lo libavahi_core_la-announce.lo libavahi_core_la-browse.lo libavahi_core_la-rrlist.lo libavahi_core_la-resolve-host-name.lo libavahi_core_la-re
ld: error: undefined reference: libintl_dgettext
>>> referenced by ../avahi-common/.libs/libavahi-common.so (disallowed by --no-allow-shlib-undefined)
ld: error: undefined reference: libintl_bindtextdomain
>>> referenced by ../avahi-common/.libs/libavahi-common.so (disallowed by --no-allow-shlib-undefined)
ld: error: undefined reference: libintl_bind_textdomain_codeset
>>> referenced by ../avahi-common/.libs/libavahi-common.so (disallowed by --no-allow-shlib-undefined)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [Makefile:1102: prioq-test] Error 1
gmake[2]: *** Waiting for unfinished jobs....
libtool: link: clang -shared  -fPIC -DPIC  .libs/libavahi_core_la-timeeventq.o .libs/libavahi_core_la-iface.o .libs/libavahi_core_la-server.o .libs/libavahi_core_la-entry.o .libs/libavahi_core_la-prioq.o .libs/libavahi_core_la-cache.o .libs/libavahi_core_la-socket.o .libs/libavahi_core_la-response-sched.o .libs/libavahi_core_la-query-sched.o .libs/libavahi_core_la-probe-sched.o .libs/libavahi_core_la-announce.o .libs/libavahi_core_la-browse.o .libs/libavahi_core_la-rrlist.o .libs/libavahi_core_la-resolve-host-name.o .libs/libavahi_core_la-resolve-address.o .libs/libavahi_core_la-browse-domain.o .libs/libavahi_core_la-browse-service-type.o .libs/libavahi_core_la-browse-service.o .libs/libavahi_core_la-resolve-service.o .libs/libavahi_core_la-dns.o .libs/libavahi_core_la-rr.o .libs/libavahi_core_la-log.o .libs/libavahi_core_la-browse-dns-server.o .libs/libavahi_core_la-fdutil.o .libs/libavahi_core_la-util.o .libs/libavahi_core_la-hashmap.o .libs/libavahi_core_la-wide-area.o .libs/libavahi_core_la-multicast-look
libtool: link: (cd ".libs" && rm -f "libavahi-core.so.7" && ln -s "libavahi-core.so.7.1.0" "libavahi-core.so.7")
libtool: link: (cd ".libs" && rm -f "libavahi-core.so" && ln -s "libavahi-core.so.7.1.0" "libavahi-core.so")
libtool: link: ar cr .libs/libavahi-core.a  libavahi_core_la-timeeventq.o libavahi_core_la-iface.o libavahi_core_la-server.o libavahi_core_la-entry.o libavahi_core_la-prioq.o libavahi_core_la-cache.o libavahi_core_la-socket.o libavahi_core_la-response-sched.o libavahi_core_la-query-sched.o libavahi_core_la-probe-sched.o libavahi_core_la-announce.o libavahi_core_la-browse.o libavahi_core_la-rrlist.o libavahi_core_la-resolve-host-name.o libavahi_core_la-resolve-address.o libavahi_core_la-browse-domain.o libavahi_core_la-browse-service-type.o libavahi_core_la-browse-service.o libavahi_core_la-resolve-service.o libavahi_core_la-dns.o libavahi_core_la-rr.o libavahi_core_la-log.o libavahi_core_la-browse-dns-server.o libavahi_core_la-fdutil.o libavahi_core_la-util.o libavahi_core_la-hashmap.o libavahi_core_la-wide-area.o libavahi_core_la-multicast-lookup.o libavahi_core_la-querier.o libavahi_core_la-addr-util.o libavahi_core_la-domain-util.o libavahi_core_la-iface-pfroute.o
libtool: link: ranlib .libs/libavahi-core.a
libtool: link: ( cd ".libs" && rm -f "libavahi-core.la" && ln -s "../libavahi-core.la" "libavahi-core.la" )
gmake[2]: Leaving directory '/home/runner/work/avahi/avahi/avahi-core'
gmake[1]: Leaving directory '/home/runner/work/avahi/avahi'
gmake[1]: *** [Makefile:841: all-recursive] Error 1
gmake: *** [Makefile:751: all] Error 2
```

--- a/.github/workflows/build.sh
+++ b/.github/workflows/build.sh
@@ -62,15 +62,12 @@ case "$1" in
         ;;
     build)
         if [[ "$OS" == FreeBSD ]]; then
-            # Do what USES="localbase:ldflags gettext-runtime gmake" do
-            # in FreeBSD Ports, namely:
+            # Do what USES="localbase:ldflags" do in FreeBSD Ports, namely:
             # - Add /usr/local/include to the compiler's search path
             # - Add /usr/local/lib to the linker's search path
-            # - Additionally link to libintl because it is a separate library,
-            #   unlike on Linux. See https://github.com/avahi/avahi/issues/726
             export CFLAGS+=" -I/usr/local/include"
             export CPPFLAGS+=" -I/usr/local/include"
-            export LDFLAGS+=" -L/usr/local/lib -lintl"
+            export LDFLAGS+=" -L/usr/local/lib"
         fi
 
         if [[ "$ASAN_UBSAN" == true ]]; then

--- a/avahi-common/Makefile.am
+++ b/avahi-common/Makefile.am
@@ -78,7 +78,7 @@ libavahi_common_la_SOURCES = \
 	i18n.c i18n.h
 
 libavahi_common_la_CFLAGS = $(AM_CFLAGS) $(PTHREAD_CFLAGS) -DAVAHI_LOCALEDIR=\"$(localedir)\"
-libavahi_common_la_LIBADD = $(AM_LDADD) $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) $(INTLLIBS)
+libavahi_common_la_LIBADD = $(AM_LDADD) $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) $(LTLIBINTL)
 libavahi_common_la_LDFLAGS = $(AM_LDFLAGS)  -version-info $(LIBAVAHI_COMMON_VERSION_INFO)
 
 strlst_test_SOURCES = \

